src/k2/tso/client/tso_clientlib.cpp

```diff
@@ -159,6 +169,17 @@ seastar::future<Timestamp> TSO_ClientLib::GetTimestampFromTSO(const TimePoint& r
         return seastar::make_exception_future<Timestamp>(TSOClientLibShutdownException());
     }
 
+    // TSO client may not yet ready (discover the tso server endpoint), let the request wait in this case.
+    if (!_readyToServe)
+    {
+        // if not ready to serve yet, wait on a new ready promise then call get this function self, as each promise can only chain one then lamda
+        K2INFO("TSO Timestamp requested when not ready to serve, request pending...");
+        _promiseReadyToServe.emplace_back(seastar::promise<>());

```

Comment:
just call emplace_back() here. no need to supply a temporary promise to emplace_back

Reply:
done.